### PR TITLE
Refactor PaymentsController for current_user and enhance validations

### PR DESCRIPTION
**Pull request by rsmoke**
This pull request enhances the payment flow by adding robust validation for payment amounts and improving the user experience around partial payments. The main changes include stricter server-side validation to prevent invalid or excessive payment amounts, improved test coverage for these scenarios, and updated UI messaging to clarify partial payment options.

**Payment Amount Validation Improvements:**

* Introduced a `validated_payment_amount` method in `PaymentsController` to ensure payment amounts are positive integers, do not exceed the user's current balance due, and are capped at a maximum of 2000. Decimal, zero, negative, or excessive values are now rejected, and users are redirected with an alert if validation fails. (`app/controllers/payments_controller.rb`, `def make_payment`, `def validated_payment_amount`, `def current_balance_due`) 

**User Experience Enhancements:**

* Added helper text to the payment form explaining that users can pay the full balance or a partial amount, improving clarity for partial payments. (`app/views/payments/payment_show.html.erb`)

**Test Coverage and Behavior:**

* Updated controller and request specs to cover new validation logic, including tests for decimal, zero, over-balance, missing, and non-numeric amounts, as well as support for both full and partial payments. (`spec/controllers/payments_controller_spec.rb`, `spec/requests/payments_spec.rb`)

### CODE REVIEW

File: app/controllers/payments_controller.rb [hunk 1]
```diff
@@ -115,6 +123,32 @@ def generate_hash(current_user, amount=100)
...
 
+      def validated_payment_amount(raw_amount)
+        amount = Integer(raw_amount, exception: false)
+        return nil if amount.nil? || amount <= 0
+
+        balance_due = current_balance_due
+        return nil if balance_due <= 0
+
+        max_amount = [balance_due.floor, MAX_PAYMENT_AMOUNT].min
+        return nil if amount > max_amount
```
[the remark this answers is not in the document]
Comment (rsmoke) [reply app/controllers/payments_controller.rb:135]: Balance and costs are always in rounded dollar amounts

File: app/views/payments/payment_show.html.erb [hunk 2]
```diff
@@ -65,6 +65,9 @@
                     within: 1..2000, required: true, class: 'form-control' %>
                 <%= f.submit "Pay Now", class: 'btn btn-sm btn-success ml-2' %>
               </div>
+              <small class="form-text text-muted mt-2">
+                The amount is prefilled with your full balance due, but you can enter a different amount to make a partial payment.
+              </small>
```
[the remark this answers is not in the document]
Comment (rsmoke) [reply app/views/payments/payment_show.html.erb:70]: Balance and costs are always in rounded dollar amounts
